packages/eslint-plugin/src/rules/no-this-alias.ts

```diff
@@ -48,11 +48,16 @@ export default util.createRule<Options, MessageIds>({
   ],
   create(context, [{ allowDestructuring, allowedNames }]) {
     return {
-      "VariableDeclarator[init.type='ThisExpression']"(
-        node: TSESTree.VariableDeclarator,
+      "VariableDeclarator[init.type='ThisExpression'], AssignmentExpression[right.type='ThisExpression']"(
+        node: TSESTree.VariableDeclarator | TSESTree.AssignmentExpression,
       ): void {
-        const { id } = node;
+        let id;
 
+        if (node.type === AST_NODE_TYPES.VariableDeclarator) {
+          id = node.id;
+        } else {
+          id = node.left;
+        }
```

Comment:
Just nitpicking, if you wanted to get a little more terse...

```suggestion
        const id =
          node.type === AST_NODE_TYPES.VariableDeclarator ? node.id : node.left;
```

Reply:
Wow, that definitely seems more readable. I was doing a lot of thinking on how to improve that part of the code, thanks for the feedback!

I must consider using more ternary expressions from now on 😅

I will add a commit to enhance this.